WalletWasabi.Gui/Controls/WalletExplorer/WalletExplorerViewModel.cs

```diff
@@ -17,51 +19,75 @@ namespace WalletWasabi.Gui.Controls.WalletExplorer
 	[Shared]
 	public class WalletExplorerViewModel : ToolViewModel, IActivatableExtension
 	{
+		private ObservableCollection<WalletViewModelBase> _wallets;
+		private ViewModelBase _selectedItem;
+
 		public override Location DefaultLocation => Location.Right;
 
 		public WalletExplorerViewModel()
 		{
 			Title = "Wallet Explorer";
 
-			_wallets = new ObservableCollection<WalletViewModel>();
+			_wallets = new ObservableCollection<WalletViewModelBase>();
+
+			WalletManager = Locator.Current.GetService<Global>().WalletManager;
 		}
 
-		private ObservableCollection<WalletViewModel> _wallets;
+		private WalletManager WalletManager { get; }
 
-		public ObservableCollection<WalletViewModel> Wallets
+		public ObservableCollection<WalletViewModelBase> Wallets
 		{
 			get => _wallets;
 			set => this.RaiseAndSetIfChanged(ref _wallets, value);
 		}
 
-		private WasabiDocumentTabViewModel _selectedItem;
-
-		public WasabiDocumentTabViewModel SelectedItem
+		public ViewModelBase SelectedItem
 		{
 			get => _selectedItem;
 			set => this.RaiseAndSetIfChanged(ref _selectedItem, value);
 		}
 
-		internal void OpenWallet(Wallet wallet, bool receiveDominant)
+		internal void OpenWallet(Wallet wallet, bool receiveDominant, bool select)
 		{
-			var walletName = Path.GetFileNameWithoutExtension(wallet.KeyManager.FilePath);
-			if (_wallets.Any(x => x.Title == walletName))
+			if (_wallets.OfType<WalletViewModel>().Any(x => x.Title == wallet.WalletName))
 			{
 				return;
 			}
 
 			WalletViewModel walletViewModel = new WalletViewModel(wallet);
-			_wallets.Add(walletViewModel);
+
+			Wallets.InsertSorted(walletViewModel);
+
+			if (select)
+			{
+				SelectedItem = walletViewModel;
+			}
+
 			walletViewModel.OpenWallet(receiveDominant);
 		}
 
+		internal void RemoveWallet(WalletViewModelBase wallet)
+		{
+			Wallets.Remove(wallet);
+		}
+
+		private void LoadWallets()
+		{
+			foreach (var walletPath in WalletManager.WalletDirectories.EnumerateWalletFiles())
+			{
+				Wallets.InsertSorted(new ClosedWalletViewModel(walletPath.FullName));
```

Comment:
Wallet should be passed as the model to the WalletViewModel.

Reply:
done